src/Arcus.Security.Core/Extensions/SecretStoreBuilderExtensions.cs

```diff
@@ -30,7 +31,29 @@ public static class SecretStoreBuilderExtensions
             Guard.For<ArgumentException>(() => !Enum.IsDefined(typeof(EnvironmentVariableTarget), target),
                 $"Requires an environment variable target of either '{EnvironmentVariableTarget.Process}', '{EnvironmentVariableTarget.Machine}', or '{EnvironmentVariableTarget.User}'");
 
-            return builder.AddProvider(new EnvironmentVariableSecretProvider(target, prefix), mutateSecretName);
+            return AddEnvironmentVariables(builder, options => options.MutateSecretName = mutateSecretName, target, prefix);
+        }
+
+        /// <summary>
+        /// Adds a secret source to the secret store of the application that gets its secrets from the environment.
+        /// </summary>
+        /// <param name="builder">The builder to create the secret store.</param>
+        /// <param name="configureOptions">The function to configure the registration of the <see cref="ISecretProvider"/> in the secret store.</param>
+        /// <param name="target">The target on which the environment variables should be retrieved.</param>
+        /// <param name="prefix">The optional prefix which will be prepended to the secret name when retrieving environment variables.</param>
+        /// <exception cref="ArgumentNullException">Thrown when the <paramref name="builder"/> is <c>null</c>.</exception>
+        /// <exception cref="ArgumentException">Thrown when the <paramref name="target"/> is outside the bounds of the enumeration.</exception>
+        public static SecretStoreBuilder AddEnvironmentVariables(
```

Comment:
Personally I think it would be simpler if both name & mutation are just a parameter here instead of having ot use options, I don't know what @fgheysels thinks?

Reply:
I already updated this (as discussed by ourselves) to parameter-approach.